src/map/Map.js

```diff
@@ -446,11 +446,14 @@ export var Map = Evented.extend({
 	setMaxBounds: function (bounds) {
 		bounds = toLatLngBounds(bounds);
 
-		if (!bounds.isValid()) {
-			this.options.maxBounds = null;
-			return this.off('moveend', this._panInsideMaxBounds);
-		} else if (this.options.maxBounds) {
+		if (this._maxBoundsSet) {
+			this._maxBoundsSet = false;
 			this.off('moveend', this._panInsideMaxBounds);
+
+			if (!bounds.isValid()) {
+				this.options.maxBounds = null;
+				return this;
+			}

```

Comment:
If the variable `bounds` is passed invalid, you don't return (like it was before) and apply it to `maxBounds` in line 459.

Reply:
@Falke-Design what does the `return this` line do then?

Reply:
If the bounds are invalid but `_maxBoundsSet` is not set, you don't come to the part `if (!bounds.isValid()) {`

Reply:
Oh, I see now, this is the case where max bounds wasn't set before.